src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -150,12 +152,110 @@ begin
   simp [e₂, direct_sum.submodule_coe, direct_sum.to_module, dfinsupp.sum_add_hom_apply]
 end
 
-/-- An orthonormal basis on a fintype `ι` for an inner product space induces an isometry with
+variables (ι) (𝕜) (E)
+
+/-- The vector given in euclidean space by being `1 : 𝕜` at coordinate `i : ι` and `0 : 𝕜` at
+all other coordinates. -/
+def euclidean_space.single {𝕜 : Type*} {ι : Type*} [fintype ι] [is_R_or_C 𝕜] (i : ι) (a : 𝕜) :
+ euclidean_space 𝕜 ι :=
+  by {classical, exact pi.single i a}
```

Comment:
I am really not an expert on decidability issues, but my understanding is that in a _definition_ (like this) you ought to assume `[decidable_eq ι]` and then make this just
```suggestion
pi.single i a
```